### PR TITLE
Change GreaterThan constraint's transform and reverse_transform methods to handle non-floats

### LINKED ISSUE #448: GreaterThan constraint's transform strategy fails on columns that are not float
### Environment Details

Please indicate the following details about the environment in which you found the bug:

* SDV version: 0.10.0
* Python version: 3.x
* Operating System: Windows, MacOS

### Error Description

Related to #421 

When a `GreaterThan` constraint is provided `transform` as the `handling_strategy`, it will fail if the `high` column is not a float. This is because the `HyperTransformer` reverse transforms the columns before the constraint's reverse transformation takes place. If the high column was initially an int or datetime, it gets transformed by the `HyperTransformer`. The constraint then has a value in the wrong form when attempting to apply its reverse transformation.

### Steps to reproduce

```
from sdv.demo import load_tabular_demo
from sdv.constraints import UniqueCombinations
from sdv.tabular import GaussianCopula
from sdv.constraints import GreaterThan

employees = load_tabular_demo()

age_gt_age_when_joined_constraint = GreaterThan(
    low='age_when_joined',
    high='age',
    handling_strategy='transform'
)
constraints = [
    age_gt_age_when_joined_constraint
]

gc = GaussianCopula(constraints=constraints)
gc.fit(employees)

sampled = gc.sample(10, conditions={'age': 40})
```

### PR DESCRIPTION
**Pull request by amontanez24**
Resolve #448

### CODE REVIEW

File: sdv/constraints/tabular.py [hunk 1]
```diff
@@ -211,14 +211,21 @@ class GreaterThan(Constraint):
...
+    self_diff_column = None
+
     def __init__(self, low, high, strict=False, handling_strategy='transform',
-                 fit_columns_model=True):
+                 fit_columns_model=True, drop=None):
         self._low = low
         self._high = high
         self._strict = strict
         self.constraint_columns = (low, high)
+        self._diff_column = f'#{self._low}#{self._high}'
```
Comment (csala) [inline sdv/constraints/tabular.py:227]: This line may be unnecessary (see comment below)

File: sdv/constraints/tabular.py [hunk 2]
```diff
@@ -230,6 +237,9 @@ def _fit(self, table_data):
                 The Table data.
         """
         self._dtype = table_data[self._high].dtype
+        self._diff_column = f'#{self._low}#{self._high}'
```
Comment (csala) [inline sdv/constraints/tabular.py:240]: Maybe we should be [finding a valid separator in a similar fashion as to what `UniqueConstraints` does](https://github.com/sdv-dev/SDV/blob/1dbbfdaca61b532ca85c9ddedde5c296b0cbccc7/sdv/constraints/tabular.py#L109)? We could move the logic to a `_find_valid_separator(table_data, columns` staticmethod at the `Constraint` class (or even `sdv.utils`), so all the constraints (and potentially other parts of SDV) can use it.
I would do either that or involve a `uuid` generated on the fly to make sure that the column name does not clash with an existing column.

File: sdv/constraints/tabular.py [hunk 3]
```diff
@@ -294,13 +308,32 @@ def reverse_transform(self, table_data):
                 Transformed data.
         """
         table_data = table_data.copy()
-        diff = (np.exp(table_data[self._high]).round() - 1).clip(0)
-        low_column = table_data[self._low]
+        diff = (np.exp(table_data[self._diff_column]).round() - 1).clip(0)
 
-        if pd.api.types.is_datetime64_ns_dtype(low_column):
-            diff = pd.to_timedelta(diff)
+        if self._drop == 'high':
```
Comment (csala) [inline sdv/constraints/tabular.py:313]: There seems to be too much duplicated code in this `if/else` block. I would attempt to move common parts to another method

File: sdv/constraints/tabular.py [hunk 4]
```diff
@@ -294,13 +308,32 @@ def reverse_transform(self, table_data):
...
+            low_column = table_data[self._low]
+            if pd.api.types.is_datetime64_ns_dtype(low_column):
+                diff = pd.to_timedelta(diff)
+
+            invalid = ~self.is_valid(table_data)
+            new_high_values = low_column.loc[invalid] + diff.loc[invalid]
+            table_data[self._high].loc[invalid] = new_high_values.astype(self._dtype)
 
-        table_data[self._high] = (low_column + diff).astype(self._dtype)
+        table_data = table_data.drop(self._diff_column, axis=1)
```
Comment (csala) [inline sdv/constraints/tabular.py:336]: a `del table_data[self._diff_column]` would also do the trick, but I do not know which one is faster and has less memory overhead. @amontanez24 could we time this and also research a bit whether one is preferable over the other one?
@pvk-developer any thoughts?

**Review by csala — CHANGES_REQUESTED** [review]
I added a few comments about things that I believe could be improved, but the overall implementation and approach seems to be very good @amontanez24 !

File: sdv/constraints/tabular.py (hunk 4 above)
Comment (amontanez24) [reply sdv/constraints/tabular.py:336]: looks like `drop` is faster https://stackoverflow.com/questions/47426089/python-del-vs-pandas-drop/51194789

**Review by csala — APPROVED** [review]
LGTM!

**Pull request merged**